paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -114,15 +105,19 @@ internal class DefaultFlowController @Inject internal constructor(
             object : LifecycleObserver {
                 @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
                 fun onCreate() {
-                    paymentController.registerLaunchersWithActivityResultCaller(
-                        activityResultCaller,
-                        ::onPaymentFlowResult
+                    paymentLauncher = paymentLauncherFactory.create(
+                        { lazyPaymentConfiguration.get().publishableKey },
```

Comment:
Does the publishableKey and stripeAccountId need to be lambdas still?

Reply:
Based on this comment:
```

Reply:
Based on this comment earlier in the file: 
```    
/**
 * [PaymentConfiguration] is [Lazy] because the client might set publishableKey and
 * stripeAccountId after creating a [DefaultFlowController].
 */
 private val lazyPaymentConfiguration: Lazy<PaymentConfiguration>,
```
There's some potential that these values can change and we'd want to keep from accessing them until needed. At least I'm assuming why the parameters for StripePaymentLauncher are lambdas. @ccen-stripe ?

Reply:
Yes, when `PaymentLauncher` is used standalone, it doesn't need lazy because the key/id are passed to its [factory method](https://github.com/stripe/stripe-android/blob/3e92b79190834dc3aab1c2d9ac2dfb7bc343afd2/payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncher.kt#L55) and is known when it's created.
However when it's used in `DefaultFlowController`, it needs to be lazy because the key/id are not passed to constructor, but [initialized separately](https://github.com/stripe/stripe-android/blob/ff31592b2847a09ec9a3e148d1a1defe8d19a099/paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/activity/BasePaymentSheetActivity.kt#L61), which can happen after constructor call.

The lambda signature in `PaymentLauncher` is to make it compatible for both cases